sub-packages/bionemo-scdl/src/bionemo/scdl/io/single_cell_memmap_dataset.py

```diff
@@ -348,13 +348,14 @@ def get_row(
             feature_vars: Optional, feature variables to extract
         Return:
             [Tuple[np.ndarray, np.ndarray]: data values and column pointes
-            pd.DataFrame: optional, corresponding features.
+            List[np.ndarray]: optional, corresponding features.
         """
         start = self.row_index[index]
         end = self.row_index[index + 1]
         values = self.data[start:end]
         columns = self.col_index[start:end]
         ret = (values, columns)
+        # ret = (np.array(values), np.array(columns)) TODO: do we want to do the conversion here? it does result in a slowdown
```

Comment:
@polinabinder1 @edawson @skothenhill-nv - would like your feedback here! I'm inclined to remove it and allow users to do the conversion from np memmap to np array on their end if needed (so for us, we would do it in get_item of Geneformer). This is because if we do the conversion here we see a slowdown from .12 to .17 seconds for calling get_row on all indices in the train split of the cellxsmall dataset.

Reply:
Update: Made the decision to do the conversion here, because returning a np.memmap array fundamentally doesn't make sense for one row of data, and the user is expecting a np array back. The tradeoff is that we accept the slowdown in the call to get_row(index).